src/hdf5_back.cc

```diff
@@ -2866,4 +3108,42 @@ Hdf5Back::VLBufToVal<std::map<std::string, std::string> >(const hvl_t& buf) {
   return x;
 }
 
+
+hvl_t Hdf5Back::VLValToBuf(const std::map<std::pair<int, std::string>, double>& x) {
+  // VL_MAP_PAIR_INT_STRING_DOUBLE implemented as VL_MAP_PAIR_INT_VL_STRING_DOUBLE
+  hvl_t buf;
+  Digest keyhash;
+  buf.len = x.size();
+  size_t itemsize = sizeof(int) + CYCLUS_SHA1_SIZE + sizeof(double);
+  size_t nbytes = itemsize * buf.len;
+  buf.p = new char[nbytes];
+  unsigned int i = 0;
```

Comment:
Here is not technically unavoidable, but it is a little silly.  `sizeof()` (http://en.cppreference.com/w/cpp/language/sizeof) returns a size_t, which is defined by the platform to be an unsigned integral type.  So making i signed would mean that the addition would have to cast one or the other of the values.